kv/consul/client_test.go

```diff
@@ -82,7 +88,10 @@ func TestWatchKeyNoRateLimit(t *testing.T) {
 }
 
 func TestReset(t *testing.T) {
-	c := NewInMemoryClient(codec.String{}, testLogger{})
+	c, closer := NewInMemoryClient(codec.String{}, testLogger{})
+	t.Cleanup(func() {
+		_ = closer.Close()
```

Comment:
```suggestion
		require.NoError(t, closer.Close())
```

_Same comment applies in other tests._

Reply:
@pracucci OK will take care of it.